src/IO/WriteBufferFromS3.h

```diff
@@ -105,6 +105,7 @@ class WriteBufferFromS3 final : public BufferWithOwnMemory<WriteBuffer>
     std::vector<String> part_tags;
 
     bool is_prefinalized = false;
+    std::atomic<bool> is_finalized = false;
```

Comment:
why is it atomic? Isn't the WriteBufferFromS3 non thread safe?

Reply:
Agree